api/python/quilt3/packages.py

```diff
@@ -784,8 +789,14 @@ def set_dir(self, lkey, path=None, meta=None):
             for f in files:
                 if not f.is_file():
                     continue
-                entry = PackageEntry(PhysicalKey.from_path(f), f.stat().st_size, None, None)
+
                 logical_key = f.relative_to(src_path).as_posix()
+
+                # check update policy
+                if update_policy == existing_policy and root.__contains__(logical_key):
```

Comment:
I don't think there's any need to explicitly check for `incoming`? since the package entry will still be created if the `logical_key` doesn't exist.
With our current use case, `incoming` update policy and non existing `logical_key` proceeds to create/update an entry.